File: kernel/scala/src/main/scala/com/twosigma/beakerx/scala/chart/xychart/plotitem/ConstantLine.scala
/*
 *  Copyright 2017 TWO SIGMA OPEN SOURCE, LLC
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF AnyVal KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package com.twosigma.beakerx.scala.chart.xychart.plotitem

import java.time.{Instant, LocalDate, LocalDateTime}
import java.util.Date

import com.twosigma.beakerx.chart.Color
import com.twosigma.beakerx.chart.xychart.plotitem.StrokeType
import com.twosigma.beakerx.scala.chart.GraphicsProperties

class ConstantLine extends com.twosigma.beakerx.chart.xychart.plotitem.ConstantLine with ConstantLineProperties {

  def this(x: Any) {
    this()
    super.setX(x.asInstanceOf[Number])
  }

  def this(x: Any, y: Any) {
    this(x)
    super.setY(y.asInstanceOf[Number])
  }

  def this(x: Any, y: Any, style: StrokeType, color: Color) {
    this(x, y)
    super.setStyle(style)
    super.setColor(color)
  }

  def this(x: Any, y: Any, color: Color, width: Float, showLabel: Boolean) {
    this(x, y)
    super.setColor(color)
    super.setWidth(width)
    super.setShowLabel(showLabel)
  }
}

trait ConstantLineProperties extends GraphicsProperties {
  this: com.twosigma.beakerx.chart.xychart.plotitem.ConstantLine =>

  def color: Option[Color] = Option(getColor)
  def color_=(c: Color): Unit = setColor(c)
  def color_=(c: java.awt.Color): Unit = setColor(c)

  def showLabel: Boolean = getShowLabel
  def showLabel_=(showLabel: Boolean): Unit = setShowLabel(showLabel)

  def style: Option[StrokeType] = Option(getStyle)
  def style_=(style: StrokeType): Unit = setStyle(style)

  def width: Float = getWidth
  def width_=(width: Float): Unit = setWidth(width)

  def x: Option[Number] = Option(getX)
  def x_=(x: Number): Unit = setX(x)
  def x_=(x: Date): Unit = setX(x)
  def x_=(x: LocalDate): Unit = setX(x)
  def x_=(x: LocalDateTime): Unit = setX(x)
  def x_=(x: Instant): Unit = setX(x)

  def y: Option[Number] = Option(getY)
  def y_=(y: Number): Unit = setY(y)
}

File: kernel/base/src/main/java/com/twosigma/beakerx/chart/xychart/plotitem/ConstantLine.java
/*
 *  Copyright 2014 TWO SIGMA OPEN SOURCE, LLC
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package com.twosigma.beakerx.chart.xychart.plotitem;

import com.twosigma.beakerx.chart.Color;
import com.twosigma.beakerx.chart.Graphics;
import com.twosigma.beakerx.util.DateUtil;

import java.time.Instant;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Calendar;
import java.util.Date;

/**
 * ConstantLine
 */
public class ConstantLine extends Graphics {
  private Number x;
  private Number y;
  private Color baseColor;
  private Float width = 1.5f;
  private StrokeType style;
  private Class plotType;
  private boolean showLabel;

  public Number getX() {
    return x;
  }

  public void setX(Number x) {
    this.x = x;
  }

  public void setX(Date x) {
    this.x = DateUtil.dateToLong(x);
  }

  public void setX(Calendar x) {
    this.x = DateUtil.dateToLong(x);
  }

  public void setX(Instant x) {
    this.x = DateUtil.dateToLong(x);
  }

  public void setX(LocalDateTime x) {
    this.x = DateUtil.dateToLong(x);
  }

  public void setX(LocalDate x) {
    this.x = DateUtil.dateToLong(x);
  }

  public Number getY() {
    return y;
  }

  public void setY(Number y) {
    this.y = y;
  }

  public void setColor(Color color) {
    this.baseColor = color;
  }

  public void setColor(java.awt.Color color) {
    setColor(new Color(color));
  }

  @Override
  public void setColori(Color color) {
    this.baseColor = color;
  }

  @Override
  public Color getColor() {
    return baseColor;
  }

  public Float getWidth() {
    return width;
  }

  public void setWidth(Float width) {
    this.width = width;
  }

  public StrokeType getStyle() {
    return style;
  }

  public void setStyle(StrokeType style) {
    this.style = style;
  }

  public Class getPlotType() {
    return plotType;
  }

  public void setPlotType(Class plotType) {
    this.plotType = plotType;
  }

  public Boolean getShowLabel() {
    return showLabel;
  }

  public void setShowLabel(boolean showLabel) {
    this.showLabel = showLabel;
  }
}
